### PR TITLE
Split code generation into separate goals for sources, test-sources, and resources.

### PR DESCRIPTION
**Pull request by motlin**
The main changes:

- Split the single maven goal into 3 based on whether it's generating sources, test-sources, or resources.
- Split the 3 input directories into 4, for api, impl, test, and services (moved services out of impl)
- Upgraded from the maven 2 style doclet to maven 3 style annotations

### CODE REVIEW

File: eclipse-collections-code-generator/src/main/java/org/eclipse/collections/codegenerator/EclipseCollectionsCodeGenerator.java [hunk 1]
```diff
@@ -16,37 +16,36 @@
...
+    private final String fileExtension;
+
     private int numFileWritten = 0;
 
-    public EclipseCollectionsCodeGenerator(String templateDirectory, File moduleBaseDir, List<URL> classPathURLs, ErrorListener errorListener)
+    public EclipseCollectionsCodeGenerator(
+            String templateDirectory,
+            List<URL> classPathURLs,
+            ErrorListener errorListener,
+            File outputDirectory, String fileExtension)
```
Comment (donraab) [inline eclipse-collections-code-generator/src/main/java/org/eclipse/collections/codegenerator/EclipseCollectionsCodeGenerator.java:42]: Move `fileExtension` to a new line.

**Review by donraab — CHANGES_REQUESTED** [review]
Hi @motlin, changes look good, although I think we can possibly remove some duplication in the Generate classes by creating an abstraction. I've recommended one small change. Please fix and rebase and then I will approve and merge. Thanks!

File: eclipse-collections-code-generator/src/main/java/org/eclipse/collections/codegenerator/EclipseCollectionsCodeGenerator.java (hunk 1 above)
Comment (motlin) [reply eclipse-collections-code-generator/src/main/java/org/eclipse/collections/codegenerator/EclipseCollectionsCodeGenerator.java]: Fixed.

Comment (motlin) [reply eclipse-collections-code-generator/src/main/java/org/eclipse/collections/codegenerator/EclipseCollectionsCodeGenerator.java]: After this lands, I have work in progress to reduce duplication here eb785181b346de9658a88315a4b2e82e065be980

**Review by donraab — APPROVED** [review]
LGTM

**Pull request merged**